osu.Desktop/Program.cs

```diff
@@ -67,18 +70,36 @@ public static void Main(string[] args)
 
                 if (!host.IsPrimaryInstance)
                 {
-                    if (args.Length > 0 && args[0].Contains('.')) // easy way to check for a file import in args
+                    if (args.Length > 0)
                     {
-                        var importer = new ArchiveImportIPCChannel(host);
+                        string arg = args[0];
 
-                        foreach (string file in args)
+                        if (arg.StartsWith(Game.OsuGameBase.OSU_PROTOCOL, StringComparison.Ordinal))
                         {
-                            Console.WriteLine(@"Importing {0}", file);
-                            if (!importer.ImportAsync(Path.GetFullPath(file, cwd)).Wait(3000))
-                                throw new TimeoutException(@"IPC took too long to send");
+                            using (new IpcServer(host).PrepareSingleUse(out OsuLinkIPCChannel handler))
+                            {
+                                Console.WriteLine(@"Handling link {0}", arg);
```

Comment:
why is this logging not in the handlers themselves?

in general all of this - detecting what the string is to pass it to a correct ipc handler, constructing it, logging - should not be in `Main()`. split a method, at least. a factory pattern may also lend well here.

Reply:
Fair enough.